cloudferrylib/os/storage/cinder_database.py

```diff
@@ -78,11 +79,21 @@ def get_client(self, params=None):
             params.cloud.tenant,
             params.cloud.auth_url)
 
-    def _update_tenant_names(self, result, tenant_id_key):
-        for entry in result:
-            tenant_name = self.identity_client.try_get_tenant_name_by_id(
-                entry[tenant_id_key], self.config.cloud.tenant)
+    def _update_tenant_names(self, entry, tenant_id_key):
```

Comment:
It seems that this method combines logic which should be split in two: 
1. One which does the actual tenant name update;
2. Other which works as predicate to identify if tenant name must be updated or not
Is there a chance to do this logical split?